org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corrections/proposals/NewCUProposal.java

```diff
@@ -0,0 +1,441 @@
+						} else {
+							setDisplayName(Messages.format(CorrectionMessages.NewCUCompletionUsingWizardProposal_createenum_inpackage_description, new String[] { typeLabel, containerLabel }));
+						}
+					}
+				} else {
+					setDisplayName(Messages.format(CorrectionMessages.NewCUCompletionUsingWizardProposal_createnewenum_inpackage_description, containerLabel));
+				}
+				break;
+			case K_ANNOTATION:
+				if (fNode != null) {
+					if (isInnerType) {
+						if (containerName.length() == 0) {
+							setDisplayName(Messages.format(CorrectionMessages.NewCUCompletionUsingWizardProposal_createinnerannotation_description, typeLabel));
+						} else {
+							setDisplayName(Messages.format(CorrectionMessages.NewCUCompletionUsingWizardProposal_createinnerannotation_intype_description, new String[] { typeLabel, containerLabel }));
+						}
+					} else {
+						if (containerName.length() == 0) {
+							setDisplayName(Messages.format(CorrectionMessages.NewCUCompletionUsingWizardProposal_createannotation_description, typeLabel));
+						} else {
+							setDisplayName(Messages.format(CorrectionMessages.NewCUCompletionUsingWizardProposal_createannotation_inpackage_description, new String[] { typeLabel, containerLabel }));
+						}
+					}
+				} else {
+					setDisplayName(Messages.format(CorrectionMessages.NewCUCompletionUsingWizardProposal_createnewannotation_inpackage_description, containerLabel));
+				}
+				break;
+			default:
+				throw new IllegalArgumentException("Unknown type kind"); //$NON-NLS-1$
+		}
+	}
+
+	private static boolean isParameterizedType(int typeKind, Name node) {
+		if (typeKind == K_CLASS || typeKind == K_INTERFACE) {
+			return node.getParent().getLocationInParent() == ParameterizedType.TYPE_PROPERTY;
+		}
+		return false;
+	}
+
+	private static String getTypeName(int typeKind, Name node) {
+		String name = ASTNodes.getSimpleNameIdentifier(node);
+
+		if (typeKind == K_CLASS || typeKind == K_INTERFACE) {
+			ASTNode parent = node.getParent();
+			if (parent.getLocationInParent() == ParameterizedType.TYPE_PROPERTY) {
+				String typeArgBaseName = getTypeArgBaseName(name);
+
+				int nTypeArgs = ((ParameterizedType) parent.getParent()).typeArguments().size();
+				StringBuilder buf = new StringBuilder(name);
+				buf.append('<');
+				if (nTypeArgs == 1) {
+					buf.append(typeArgBaseName);
+				} else {
+					for (int i = 0; i < nTypeArgs; i++) {
+						if (i != 0) {
+							buf.append(", "); //$NON-NLS-1$
+						}
+						buf.append(typeArgBaseName).append(i + 1);
+					}
+				}
+				buf.append('>');
+				return buf.toString();
+			}
+		}
+		return name;
+	}
+
+	private TypeDeclaration findEnclosingTypeDeclaration(ASTNode node, String typeName) {
+		Iterator<ASTNode> iter;
+		if (node instanceof CompilationUnit) {
+			iter = ((CompilationUnit) node).types().iterator();
+		} else if (node instanceof TypeDeclaration) {
+			if (typeName.equals(((TypeDeclaration) node).getName().toString())) {
```

Comment:
Use `Objects.equals()`, it's able to deal with corner case better.

Reply:
sure.